[C10D] Reduce default timeout from 30m to 10m

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110910
* __->__ #110901

This timeout affects NCCL/Gloo/UCC backends (and potentially any other
backend depending on default options).

The 30 minute timeout is causing jobs to waste valuable cluster time.

It's not clear how low we can set the timeout.  In practice, collectives
generally are expected to complete in milliseconds or seconds, but
certain exceptional cases may exist.  This change is an attempt
to make a significant improvement in a low-risk way.

If you are a fringe use case and know that you expect/need a long
timeout, please consider explicitly overriding the default timeout
to a value appropriate to your application.

wconstab committed:
[C10D] Reduce default timeout from 30m to 10m

This timeout affects NCCL/Gloo/UCC backends (and potentially any other
backend depending on default options).

The 30 minute timeout is causing jobs to waste valuable cluster time.

It's not clear how low we can set the timeout.  In practice, collectives
generally are expected to complete in milliseconds or seconds, but
certain exceptional cases may exist.  This change is an attempt
to make a significant improvement in a low-risk way.

If you are a fringe use case and know that you expect/need a long
timeout, please consider explicitly overriding the default timeout
to a value appropriate to your application.

[ghstack-poisoned]

fduwjj commented on torch/csrc/distributed/c10d/Backend.hpp:
```diff
@@ -17,7 +17,7 @@
 #include <torch/csrc/distributed/c10d/debug.h>
 
 constexpr auto kBackendDefaultTimeout =
-    std::chrono::milliseconds(30 * 60 * 1000);
```
Actually I am thinking of whether we can make this configurable by users? I mean 10 mins sound make sense to me but what if users want it to stay alive longer so that they can gdb into the host and debug?

wconstab replied:
oh, it is already configurable via `dist.init_process_group(, timeout=<timeout>)`

But there is no env, which is a possible gap we could address.  I'm not sure if in practice trainers expose an ENV.

Should we just make it an ENV now?

fduwjj replied:
Hmm Got it, then it's trainer's responsibility to specify this. I didn't know it is already configurable.

xw285cornell replied:
just wondering - will it be "safer" if we make the 10min change just for nccl, to avoid breaking things like cpu training which uses gloo?

wconstab replied:
Yea, if you think cpu training needs 30min timeout then I can change just the nccl timeout.

wconstab committed:
Update on "[C10D] Reduce default timeout from 30m to 10m"


This timeout affects NCCL/Gloo/UCC backends (and potentially any other
backend depending on default options).

The 30 minute timeout is causing jobs to waste valuable cluster time.

It's not clear how low we can set the timeout.  In practice, collectives
generally are expected to complete in milliseconds or seconds, but
certain exceptional cases may exist.  This change is an attempt
to make a significant improvement in a low-risk way.

If you are a fringe use case and know that you expect/need a long
timeout, please consider explicitly overriding the default timeout
to a value appropriate to your application.

[ghstack-poisoned]

fduwjj reviewed (approved):
LGTM

wconstab commented:
Closing in favor of https://github.com/pytorch/pytorch/pull/110947
Might be riskier to change the default of the non-nccl groups
